Дополнительная функциональность

KlimovAndreyN opened the pull request:
Дополнительная функциональность уже реализована

---
:mortar_board: [Дополнительная функциональность](https://up.htmlacademy.ru/ecmascript/23/user/2486639/tasks/18)

KlimovAndreyN committed:
Дополнительная функциональность уже реализована

KlimovAndreyN committed:
сообщение об ошибке загрузке

KlimovAndreyN committed:
опечатка

KlimovAndreyN committed:
комментарии

fyvfyv commented on src/const.js:
```diff
@@ -33,15 +33,15 @@ const filterEmptyMessage = {
 };
 
-const SortingType = {
+const SortingType = { //? columns + canSort или что то другое
```
SortingType выглядит ок

KlimovAndreyN replied:
ок

fyvfyv commented on src/model/events-model.js:
```diff
@@ -4,9 +4,9 @@ import { UpdateType } from '../const.js';
   #eventsApiService = null;
-  #destinations = [];
+  #destinations = []; //? Set/Map? нужны методы доступа по id и name
   #offers = new Map();
-  #events = [];
+  #events = []; //? Set/Map? нужны методы доступа по id
```
Методы поиска можно реализовать и на массивах.
С Set/Map надо учесть ковертацию в них и обратно при работе с бэкендом

KlimovAndreyN replied:
ок

fyvfyv commented on src/presenter/filter-presenter.js:
```diff
@@ -37,9 +37,11 @@ export default class FilterPresenter {
   #onEventsModelChange = (updateType) => {
-    if (updateType !== UpdateType.PATCH) {
-      this.init();
+    if ([UpdateType.PATCH, UpdateType.INIT_ERROR].includes(updateType)) {
```
А в каком случае можно попасть в ситуацию, когда `onEventsModelChange` сработает при `INIT_ERROR`?

fyvfyv commented on src/presenter/trip-presenter.js:
```diff
@@ -149,6 +149,11 @@ export default class TripPresenter {
+      case UpdateType.INIT_ERROR:
+        this.#isLoading = false;
+        this.#removeLoading();
+        render(new MessageView({ message: MessageType.FAILEAD }), this.#tripEventsElement); // удаление и #isFailedLoading нет т.к. только возможно нажать обновить страницу
```
И константу надо обновить

```suggestion
        render(new MessageView({ message: MessageType.FAILED }), this.#tripEventsElement); // удаление и #isFailedLoading нет т.к. только возможно нажать обновить страницу
```

KlimovAndreyN replied:
переделал на Model.init.catch

fyvfyv reviewed (commented):
Сортировка не работает

KlimovAndreyN committed:
восстановил сортировку и фильтрацию.
убрал включенные фильтры по умолчанию, добавлял изза маркапов, а авто тесты требуют другого поведения.

KlimovAndreyN committed:
переработал показ сообщения при начальной загрузке

KlimovAndreyN committed:
filterPresenter подписал на обновления filterModel.
заметки и комментарии

KlimovAndreyN committed:
исправления после автотестов

KlimovAndreyN committed:
lint errror

KlimovAndreyN committed:
автотесты без ошибок

KlimovAndreyN committed:
убрал вызов лишнего расчета в 'get template'

KlimovAndreyN committed:
убрал лишние вычисления в "get template"

KlimovAndreyN committed:
исправил названия, немного рефакторинга

KlimovAndreyN committed:
фильтруем события один раз при изменении моделей (MAJOR и MINOR).
Фильтрация и доступные фильтры определяются на одну и ту же дату.
Авто тесты при этом работают!

Добавил отдельные обработчики изменений, которые вызывают единый метод.

KlimovAndreyN committed:
перенес вызов he.encode в отображение.
рефакторинг.

KlimovAndreyN committed:
Merge branch 'htmlacademy-ecmascript:master' into module8-task3

KlimovAndreyN committed:
критерии

KlimovAndreyN committed:
докоммит

KlimovAndreyN committed:
рефокторинг

KlimovAndreyN committed:
Delete src/utils/utils.js

KlimovAndreyN committed:
Delete src/utils/sorting.js

KlimovAndreyN committed:
подправил названия

KlimovAndreyN committed:
скорректировал названия, убрал неиспользуемый код

KlimovAndreyN committed:
прогнать автотесты

рефакторинг, немного изменил отрисовку событий

KlimovAndreyN committed:
автотесты ок. скорректировал вопросы и заметки

KlimovAndreyN committed:
'-' -> &mdash;

KlimovAndreyN committed:
перегруппировал порядок отрисовки

Теперь, при изменении сортировки меняется только отображение событий

KlimovAndreyN committed:
рефакториг

KlimovAndreyN committed:
критерии
